GooglePlayInstant/Editor/QuickDeploy/LoadingScreenGenerator.cs

```diff
@@ -68,46 +71,64 @@ public static void GenerateLoadingScreenScene(string assetBundleUrl, string load
             }
 
             // Removes the loading scene if it is present, otherwise does nothing.
-            EditorSceneManager.CloseScene(SceneManager.GetSceneByName(LoadingSceneName), true);
+            EditorSceneManager.CloseScene(SceneManager.GetSceneByName(Path.GetFileNameWithoutExtension(SceneName)), true);
 
             var loadingScreenScene = EditorSceneManager.NewScene(NewSceneSetup.EmptyScene, NewSceneMode.Additive);
 
-            Directory.CreateDirectory(LoadingScreenResourcesPath);
+            Directory.CreateDirectory(ResourcesDirectoryPath);
 
-            GenerateLoadingScreenConfigFile(assetBundleUrl, LoadingScreenJsonPath);
+            GenerateConfigFile(assetBundleUrl, JsonFilePath);
 
-            var loadingScreenGameObject = new GameObject(LoadingScreenCanvasName);
+            var loadingScreenGameObject = new GameObject(CanvasName);
 
-            AddLoadingScreenImageToScene(loadingScreenGameObject, loadingScreenImagePath);
+            AddImageToScene(loadingScreenGameObject, loadingScreenImagePath);
 
-            AddLoadingScreenScript(loadingScreenGameObject);
+            AddScript(loadingScreenGameObject);
 
-            LoadingBar.AddLoadingScreenBarComponent(loadingScreenGameObject);
+            LoadingBar.AddComponent(loadingScreenGameObject);
 
-            bool saveOk = EditorSceneManager.SaveScene(loadingScreenScene,
-                Path.Combine(LoadingScreenScenePath, LoadingSceneName + ".unity"));
+            bool saveOk = EditorSceneManager.SaveScene(loadingScreenScene, SceneFilePath);
 
             if (!saveOk)
             {
                 // Not a fatal issue. User can attempt to resave this scene.
                 var warningMessage = string.Format("Issue while saving scene {0}.",
-                    LoadingSceneName);
-                
+                    SceneName);
+
                 Debug.LogWarning(warningMessage);
-                
-                DialogHelper.DisplayMessage(LoadingScreenSaveErrorTitle, warningMessage);
+
+                DialogHelper.DisplayMessage(SaveErrorTitle, warningMessage);
             }
+            else
+            {
+                //TODO: investigate GUI Layout errors that occur when moving this to DialogHelper
+                if (EditorUtility.DisplayDialog("Change Main Scene in Build?",
+                    "Would you like to change the main scene in build to the loading screen?", "Change", "Do Not Change"))
```

Comment:
"Would you like to replace any existing Scenes in Build with the loading screen scene?"

I'm ok with "Change"/"Do Not Change", but slightly prefer Yes/No.